cmd/tailscale/cli/file.go

```diff
@@ -286,22 +287,109 @@ func runCpTargets(ctx context.Context, args []string) error {
+	createNumberedFiles onConflict = "rename"    //  Create an alternately named file in the style of Chrome Downloads
+)
+
+func (v *onConflict) String() string {
+	return string(*v)
+}
+
+func (v *onConflict) Set(s string) error {
+	if s == "" {
+		*v = skipOnExist
+		return nil
+	}
+	*v = onConflict(strings.ToLower(s))
+	if *v != skipOnExist && *v != overwriteExisting && *v != createNumberedFiles {
+		return fmt.Errorf("'%s' not one of (skip|overwrite|rename)", s)
+	}
+	return nil
+}
+
 var fileGetCmd = &ffcli.Command{
 	Name:       "get",
-	ShortUsage: "file get [--wait] [--verbose] <target-directory>",
+	ShortUsage: "file get [--wait] [--verbose] [--conflict=(skip|overwrite|rename)] <target-directory>",
 	ShortHelp:  "Move files out of the Tailscale file inbox",
 	Exec:       runFileGet,
 	FlagSet: (func() *flag.FlagSet {
 		fs := newFlagSet("get")
 		fs.BoolVar(&getArgs.wait, "wait", false, "wait for a file to arrive if inbox is empty")
 		fs.BoolVar(&getArgs.verbose, "verbose", false, "verbose output")
+		fs.Var(&getArgs.conflict, "conflict", "behavior when target file already exists")
 		return fs
 	})(),
 }
 
-var getArgs struct {
-	wait    bool
-	verbose bool
+var getArgs = struct {
+	wait     bool
+	verbose  bool
+	conflict onConflict
+}{conflict: skipOnExist}
+
+func numberedFileName(dir, name string, i int) string {
+	extension := path.Ext(name)
+	numberedFileName := fmt.Sprintf("%s (%d)%s", name[:len(name)-len(extension)], i, extension)
+	return filepath.Join(dir, numberedFileName)
+}
+
+func openFileOrSubstitute(dir, base string, action onConflict) (of *os.File, err error) {
+	targetFile := filepath.Join(dir, base)
+	of, openError := os.OpenFile(targetFile, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0644)
+	if openError != nil {
+		switch action {
+		default:
+			return nil, fmt.Errorf("file issue. how to resolve this conflict? no one knows.")
+		case skipOnExist:
+			if _, err = os.Stat(targetFile); err == nil {
+				err = fmt.Errorf("refusing to overwrite %v. %w", targetFile, openError)
+			}
+			err = fmt.Errorf("failed to write; %w", openError)
+		case overwriteExisting:
+			if of, err = os.OpenFile(targetFile, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644); err != nil {
+				err = fmt.Errorf("unable to overwrite %v. %w", targetFile, err)
+			}
+		case createNumberedFiles:
+			// It's possible the target directory or filesystem isn't writable by us,
+			// not just that the target file(s) already exists.  For now, give up after
+			// a limited number of attempts.  In future, maybe distinguish this case
+			// and follow in the style of https://tinyurl.com/chromium100
+			maxAttempts := 100
+			for i := 1; i < maxAttempts; i++ {
+				targetFile = numberedFileName(dir, base, i)
+				if of, err = os.OpenFile(targetFile, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0644); err == nil {
+					return
+				}
+			}
+			err = fmt.Errorf("unable to find a name for writing %v.  original error: %w", targetFile, openError)
+		}
+	}
+	return
```

Comment:
sorry, no naked returns in funcs this long. Only when it's a few lines long at most.

Reply:
done, good to know the convention. probably clearer this way.